mcstas-comps/union/Union_master.comp

```diff
@@ -1265,7 +1265,7 @@ TRACE
               printf("Checking if child %d of current_volume = %d have intersections. \n",*check,current_volume);
             #endif
             // Only check the first of the two results in the intersection table, as they are ordered, and the second is of no interest
-            if (intersection_time_table.calculated[*check] == 1 && intersection_time_table.intersection_times[*check][0] > time_propagated_without_scattering) {
+            if (intersection_time_table.calculated[*check] == 1 && intersection_time_table.intersection_times[*check][0] > time_propagated_without_scattering && !Volumes[current_volume]->skip_hierarchical_optimization) {
```

Comment:
Here the information skip_hierachical_optimization is placed on the volume structure, since it's related to the geometry, I think it's nicer to place it on the geometry structure. This is also easier to access in the geometry components.

The check here in master can be moved outside of the for loop as it doesn't relate to the children's of the current volume, but just the volume itself, so it can just be an if from line 1262 to 1304, which would prevent intersection_with_children being set to 1 and thus the intersection call will be made.